editors/code/src/main.ts

```diff
@@ -48,21 +48,12 @@ export async function activate(context: vscode.ExtensionContext) {
     ctx = await Ctx.create(config, context, serverPath);
 
     // Commands which invokes manually via command palette, shortcut, etc.
-    ctx.registerCommand('reload', (ctx) => {
-        return async () => {
-            vscode.window.showInformationMessage('Reloading rust-analyzer...');
-            // @DanTup maneuver
-            // https://github.com/microsoft/vscode/issues/45774#issuecomment-373423895
-            await deactivate();
-            for (const sub of ctx.subscriptions) {
-                try {
-                    sub.dispose();
-                } catch (e) {
-                    log.error(e);
-                }
-            }
-            await activate(context);
-        };
```

Comment:
I'd rather keep this loop here, than wrap it into an abstraciton. This is another case where we do a horrible side-effectful thing, so not having to goto def to figure the control flow helps quite a bit. From the issue comment, seems like we need just to do `ctx.subscriptions = []`?

Reply:
`subscriptions` is marked as read-only, so it cannot be reassigned

Reply:
```
            while (ctx.subscriptions.length > 0) {
                const sub = ctx.subscriptions.pop()!
                try {
                    sub.dispose();
                } catch (e) {
                    log.error(e);
                }
            }
```